pkg/ingester/flush.go

```diff
@@ -333,6 +337,10 @@ func (i *Ingester) flushChunks(ctx context.Context, fp model.Fingerprint, metric
 
 	wireChunks := make([]chunk.Chunk, 0, len(chunkDescs))
 	for _, chunkDesc := range chunkDescs {
+		// Do not write chunks below the minimum chunk duration.
+		if chunkDesc.C.Len() < i.cfg.MinChunkLength {
```

Comment:
nit: `Len()` is expensive on most chunk types (not bigchunk), so you could re-use the value 15 lines later

Reply:
I ended up moving the code around in order to only drop idle chunks that have been closed. I don't think it is as easy to reuse the value anymore.